influunt-api/app/models/ControladorFisico.java

```diff
@@ -0,0 +1,117 @@
+import json.deserializers.InfluuntDateTimeDeserializer;
+import json.serializers.InfluuntDateTimeSerializer;
+import org.joda.time.DateTime;
+
+import javax.persistence.*;
+import javax.validation.Valid;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * Created by lesiopinheiro on 8/16/16.
+ */
+@Entity
+@Table(name = "controladores_fisicos")
+public class ControladorFisico extends Model implements Serializable {
+
+    public static Finder<UUID, ControladorFisico> find = new Finder<UUID, ControladorFisico>(ControladorFisico.class);
+
+    @Id
+    private UUID id;
+
+    @Column
+    private String idJson;
+
+    @OneToMany(mappedBy = "controladorFisico", cascade = CascadeType.ALL)
+    @Valid
+    private List<VersaoControlador> versoes;
+
+    @Column
+    @JsonDeserialize(using = InfluuntDateTimeDeserializer.class)
+    @JsonSerialize(using = InfluuntDateTimeSerializer.class)
+    @CreatedTimestamp
+    private DateTime dataCriacao;
+
+    @Column
+    @JsonDeserialize(using = InfluuntDateTimeDeserializer.class)
+    @JsonSerialize(using = InfluuntDateTimeSerializer.class)
+    @UpdatedTimestamp
+    private DateTime dataAtualizacao;
+
+
+    public UUID getId() {
+        return id;
+    }
+
+    public void setId(UUID id) {
+        this.id = id;
+    }
+
+    public String getIdJson() {
+        return idJson;
+    }
+
+    public void setIdJson(String idJson) {
+        this.idJson = idJson;
+    }
+
+    public List<VersaoControlador> getVersoes() {
+        return versoes;
+    }
+
+    public void setVersoes(List<VersaoControlador> versoes) {
+        this.versoes = versoes;
+    }
+
+    public DateTime getDataCriacao() {
+        return dataCriacao;
+    }
+
+    public void setDataCriacao(DateTime dataCriacao) {
+        this.dataCriacao = dataCriacao;
+    }
+
+    public DateTime getDataAtualizacao() {
+        return dataAtualizacao;
+    }
+
+    public void setDataAtualizacao(DateTime dataAtualizacao) {
+        this.dataAtualizacao = dataAtualizacao;
+    }
+
+    public Controlador getControladorAtivo() {
+        VersaoControlador versaoControlador = VersaoControlador.find.fetch("controlador").where()
+                .and(Expr.eq("controlador_fisico_id", this.id.toString()), Expr.eq("status_versao_controlador", StatusVersaoControlador.ATIVO)).findUnique();
+        return (versaoControlador != null) ? versaoControlador.getControlador() : null;
+    }
+
+    public Controlador getControladorAtivoOuEditando() {
+        VersaoControlador versaoControlador = this.getVersoes().stream().filter(versaoControladorAux ->
+                StatusVersaoControlador.ATIVO.equals(versaoControladorAux.getStatusVersaoControlador()) ||
+                        StatusVersaoControlador.EDITANDO.equals(versaoControladorAux.getStatusVersaoControlador())).findFirst().orElse(null);
+
```

Comment:
Quando um `Controlador` for clonado, ele mudará de status ativo para `ARQUIVADO`, sendo assim, não irá ocorrer o risco de haver duas versões do mesmo controlador no status `Ativo`e `Editando`.